File: c++/LinearDataStructures/DataStructures/Node.cpp
#include "stdafx.h"
#include "Node.h"

template<typename T>
Node<T>::Node(T data, Node<T>* next) {
	this->data = data;
	this->next = next;
}


template<typename T>
Node<T>::~Node() {
//	delete &data;
	delete next;
}

template<typename T>
bool Node<T>::operator==(Node<T>& other) {
	return other.data == data && other.next == next;
}

template<typename T>
bool Node<T>::operator!=(Node<T>& other) {
	return !(*this == other);
}

File: Java/PersistentDataStructures/src/Node.java

public class Node<T> {

	private T data;
	private Node<T> next;
	
	public Node(T data, Node<T> next){
		this.data = data;
		this.next = next;
	}
	
	public Node<T> next(){
		return next;
	}
	
	public T data(){
		return data;
	}
}
